assets/src/pages/room/components/StreamPlayer/UnpinnedTilesSection.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { FC, useMemo } from "react";
+import MediaPlayerTile from "./MediaPlayerTile";
+import { MembraneWebRTC, TrackEncoding } from "@jellyfish-dev/membrane-webrtc-js";
+import clsx from "clsx";
+import { MediaPlayerTileConfig, TrackWithId } from "../../../types";
+import PeerInfoLayer from "./PeerInfoLayer";
+import { GridConfigType, getGridConfig } from "../../../../features/room-page/utils/getVideoGridConfig";
+import NameTag from "../../../../features/room-page/components/NameTag";
+import InitialsImage from "../../../../features/room-page/components/InitialsImage";
+import { PinTileLayer } from "../../../../features/room-page/components/PinComponents";
+import {
+  DisabledMicIcon,
+  isLoading,
+  showDisabledIcon,
+} from "../../../../features/room-page/components/DisabledTrackIcon";
+
+const getGridStyle = (
+  gridConfig: GridConfigType,
+  oneColumn: boolean,
+  videoInVideo: boolean,
+  fixedRatio: boolean
+): string => {
+  if (!oneColumn)
+    return clsx(
+      "h-full w-full",
+      gridConfig.columns,
+      gridConfig.grid,
+      gridConfig.gap,
+      gridConfig.padding,
+      gridConfig.rows
+    );
+  if (fixedRatio)
+    return clsx(
+      "w-[400px]",
+      videoInVideo
+        ? "h-[220px] absolute bottom-4 right-4 z-10"
+        : "h-full flex flex-wrap flex-col content-center justify-center"
+    );
+
+  return "h-full w-full grid flex-1 grid-flow-row auto-rows-fr grid-cols-1 gap-y-3";
+};
+
+type Props = {
+  tileConfigs: MediaPlayerTileConfig[];
+  showSimulcast?: boolean;
+  selectRemoteTrackEncoding?: (peerId: string, trackId: string, encoding: TrackEncoding) => void;
+  oneColumn: boolean; // screensharing or pinned user
+  webrtc?: MembraneWebRTC;
+  pin: (tileId: string) => void;
+  videoInVideo: boolean;
+  blockPinning: boolean;
+};
+
+const UnpinnedTilesSection: FC<Props> = ({
+  tileConfigs,
+  showSimulcast,
+  oneColumn,
+  webrtc,
+  pin,
+  videoInVideo,
+  blockPinning,
+}: Props) => {
+  const gridConfig = getGridConfig(tileConfigs.length);
+
+  const videoGridStyle = useMemo(
+    () => getGridStyle(gridConfig, oneColumn, videoInVideo, tileConfigs.length === 1),
+    [gridConfig, oneColumn, videoInVideo, tileConfigs.length]
+  );
+  const tileStyle = !oneColumn
+    ? clsx(gridConfig.span, gridConfig.tileClass)
+    : tileConfigs.length === 1
+    ? "w-[400px] h-[220px]"
+    : "";
```

Comment:
It's hard for me to reason about nested ternary operators. Could you extract it to function with well-defined simple ifs?

Reply:
I've left it as it is. I think it is readable enough, especially if well formatted.